Switch from gometalinter to golangci-lint

gpaul opened the pull request:
The golangci-lint linter is less resource intensive than gometalinter. Also, gometalinter has been deprecated in favour of golangci-lint.

Hopefully this fixes a rare linting timeout when the CI server is busy.

Fixes https://jira.mesosphere.com/browse/DCOS_OSS-2415

bbannier commented on Dockerfile:
```diff
@@ -37,12 +37,10 @@ ENV GOPATH /go
-    go get -u golang.org/x/tools/cmd/goimports && \
+RUN mkdir -p /go/src/github.com/golangci/ && \
+    cd /go/src/github.com/golangci && \
+    git clone https://github.com/golangci/golangci-lint.git --branch=v1.17.1 && \
```
Let's move the version number into a constant like we do for other dependencies for easier maintenance.

bbannier commented on pkg/csilvm/csilvm_test.go:
```diff
@@ -1,5 +1,6 @@
 // +build !unit
 
+//nolint:errcheck
```
I feel we should also add a `TODO` here ;)

bbannier replied:
I like `try` and `check` much more than a `TODO`.

bbannier commented on pkg/lvm/lvm_test.go:
```diff
@@ -1,5 +1,6 @@
 // +build !unit
 
+//nolint:errcheck
```
`TODO`?

bbannier commented on pkg/lvm/lvm.go:
```diff
@@ -859,8 +859,14 @@ func run(cmd string, v interface{}, extraArgs ...string) error {
-		lvmlock.Lock()
-		defer lvmlock.Unlock()
+		if lerr := lvmlock.Lock(); lerr != nil {
+			return lerr
```
Looking at the impl of flock the only possible errors we might get would be related to failures using the configured file. Does it make sense to decorate this error with that additional context so users see e.g., `Could not lock configured file lock: ...`?

gpaul commented:
@nfnt @bbannier I've addressed all comments. I've introduced  two functions: `check` which panics when `err != nil` and `try`, which just logs the error. The `try` function is necessary because if a test reaches the greater test cleanup function, it performs its own cleanup but we also want to defer cleanup of resources as we create them, in case a subsequent step fails before the greater test cleanup is reached and scheduled.

gpaul commented:
I'll squash the commits once the PR is approved.
